src/SFA.DAS.Employer.PR.Web/Controllers/YourTrainingProvidersController.cs

```diff
@@ -15,26 +15,45 @@ namespace SFA.DAS.Employer.PR.Web.Controllers;
 [Route("accounts/{employerAccountId}/providers", Name = RouteNames.YourTrainingProviders)]
 public class YourTrainingProvidersController(IOuterApiClient _outerApiClient, ISessionService _sessionService) : Controller
 {
+    [HttpGet]
     public async Task<IActionResult> Index([FromRoute] string employerAccountId, CancellationToken cancellationToken)
     {
         _sessionService.Delete<AddTrainingProvidersSessionModel>();
 
-        var legalEntities = await _outerApiClient.GetAccountLegalEntities(employerAccountId, cancellationToken);
-        YourTrainingProvidersViewModel model = InitialiseViewModel(legalEntities);
+        var response = await _outerApiClient.GetAccountLegalEntities(employerAccountId, cancellationToken);
+        var accountLegalEntities = response.AccountLegalEntities;
+
+        YourTrainingProvidersViewModel model = InitialiseViewModel(accountLegalEntities);
```

Comment:
This should be determined in the SelectLegalEntityController so that you retrieve this only once if there are multiple legal entities.

Reply:
I think the comment has moved after your changes were removed from here. So you have done the expected work.